packages/hebao_v1/contracts/modules/transfers/QuotaTransfers.sol

```diff
@@ -194,19 +89,17 @@ contract QuotaTransfers is TransferModule
         onlyWhenWalletUnlocked(wallet)
         onlyFromMetaTxOrWalletOwner(wallet)
     {
-        (bool whitelisted,) = controller.whitelistStore().isWhitelisted(wallet, to);
-        if (whitelisted) {
-            approveInternal(wallet, token, to, amount);
-            return;
-        }
-
         uint allowance = ERC20(token).allowance(wallet, to);
         if (allowance >= amount) {
-            approveInternal(wallet, token, to, amount);
             return;
         }
 
-        updateQuota(wallet, token, amount.sub(allowance));
+        (bool whitelisted,) = controller.whitelistStore().isWhitelisted(wallet, to);
+        if (!whitelisted) {
+            updateQuota(wallet, token, amount.sub(allowance));
+        }
+
+        approveInternal(wallet, token, to, 0);
```

Comment:
Was this added for non-standard ERC20 behaviour where the amount approved is added instead of overwritten?

Reply:
Yes, In some token contract, the allowance can not be overwritten if it is not zero.